#!/usr/bin/env python
PACKAGE = "screw_detection_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#list of parameters
#gen.add('pixel_per_centimeter', int_t, 0, "Pixels peer centimeter on screw plane", 1, 0, 1000)
gen.add('contour_threshold', int_t, 0, "Threshold for screw contour detection", 150, 0, 300)
gen.add('contour_threshold_table', int_t, 0, "Threshold for table contour detection", 160, 0, 300)
gen.add('screw_size_cm', double_t, 0, "Size of screw head holes in cm", 1.4, 0 , 10)
gen.add('table_size_cm', double_t, 0, "Size of table in cm", 30, 0 , 100)
gen.add('screw_radius_cm', double_t, 0, "Radius on which screws are located", 21, 0 , 100)
gen.add('accumulator_threshold', int_t, 0, "Threshold for screw accumulator", 14, 0, 200)
gen.add('accumulator_threshold_table', int_t, 0, "Threshold for table accumulator", 20, 0, 200)

exit(gen.generate(PACKAGE, "screw_detection_ros", "ScrewDetectorParams"))
